scripts/lib/CIME/namelist.py

```diff
@@ -112,7 +112,7 @@
 
 # Fortran syntax regular expressions.
 # Variable names.
-FORTRAN_NAME_REGEX = re.compile(r"^[a-z][a-z0-9_]{0,62}$", re.IGNORECASE)
+FORTRAN_NAME_REGEX = re.compile(r"^[a-z][a-z0-9_]{0,62}(\((\d+)\))?$", re.IGNORECASE)
```

Comment:
What about `foo(-3)` or `foo(+3)`, both of which are also valid?
I have a regex for that which also includes (integer constant) array sections if you are interested.
As a side note, since you use `re.IGNORECASE`, can we get rid of all the scary language in namelist_definition_drv.xml about the id having to be lower case?